packages/components/src/experience-control/index.tsx

```diff
@@ -1,97 +1,126 @@
+			value={ value }
+			name={ name }
+			aria-label={ ariaLabel }
+		/>
+		<div className="experience-control__option-content">{ children }</div>
+	</label>
 );
 
-const ExperienceControlBase = ( {
-	label,
-	children,
-	helpText,
-}: {
-	label: string;
+interface ExperienceControlBaseProps {
 	children: React.ReactNode;
-	helpText?: string;
-} ) => (
-	<div className="experience-control">
-		<div className="experience-control__label">{ label }</div>
-		<div className="experience-control__buttons">{ children }</div>
-		{ helpText && <div className="experience-control__help-text">{ helpText }</div> }
-	</div>
-);
+	label?: string;
+	help?: string;
+}
+
+const ExperienceControlBase = ( { children, ...props }: ExperienceControlBaseProps ) => {
+	const { baseControlProps } = useBaseControlProps( props );
+
+	return (
+		<BaseControl className="experience-control" { ...baseControlProps }>
+			<div className="experience-control__options" role="radiogroup">
+				{ children }
+			</div>
+		</BaseControl>
+	);
+};
 
-export function ExperienceControl( {
-	label,
-	onChange,
-	selectedExperience,
-	helpText,
-}: {
+interface ExperienceControlProps {
 	label: string;
-	onChange: ( experience: string ) => void;
-	selectedExperience: string;
-	helpText?: string;
-} ) {
-	const handleChange = ( experience: string ) => {
-		onChange( experience );
-	};
+	onChange: ( experience: ExperienceType ) => void;
+	value: ExperienceType;
+	help?: string;
+}
+
+export function ExperienceControl( { label, onChange, value, help }: ExperienceControlProps ) {
+	const translate = useTranslate();
 
-	const options = [
+	const options: ExperienceOption[] = [
 		{
-			value: Experience.GOOD,
+			value: ExperienceType.GOOD,
 			icon: <IconGood />,
+			ariaLabel: translate( 'Rate as good experience' ),
 		},
 		{
-			value: Experience.NEUTRAL,
+			value: ExperienceType.NEUTRAL,
 			icon: <IconNeutral />,
+			ariaLabel: translate( 'Rate as neutral experience' ),
 		},
 		{
-			value: Experience.BAD,
+			value: ExperienceType.BAD,
 			icon: <IconBad />,
+			ariaLabel: translate( 'Rate as bad experience' ),
 		},
 	];
 
+	const radioGroupName = `experience-control-${ label.toLowerCase().replace( /\s+/g, '-' ) }`;
+
 	return (
-		<ExperienceControlBase label={ label } helpText={ helpText }>
+		<ExperienceControlBase label={ label } help={ help }>
 			{ options.map( ( option ) => (
 				<ExperienceControlOption
 					key={ option.value }
 					className={ `is-${ option.value }` }
-					isSelected={ selectedExperience === option.value }
-					onClick={ () => handleChange( option.value ) }
+					checked={ value === option.value }
+					onClick={ () => onChange( option.value ) }
+					value={ option.value }
+					name={ radioGroupName }
+					ariaLabel={ option.ariaLabel }
 				>
 					{ option.icon }
 				</ExperienceControlOption>
 			) ) }
 		</ExperienceControlBase>
 	);
 }
+
 ExperienceControl.Base = ExperienceControlBase;
 ExperienceControl.Option = ExperienceControlOption;
```

Comment:
From what I gathered from the Figma thread, we don't have a good idea of what abstractions will be appropriate yet. We'll want to work closely with the people needing the initial use cases, so they don't start customizing it locally. Let's keep the subcomponents internal for now?

```suggestion
```